torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -31,7 +31,8 @@ def _create_label(self, edge_label_index, edge_label):
             return edge_label_index, edge_label
 
         if edge_label is None:
-            edge_label = torch.ones(num_pos_edges, device=device)
+            edge_label = torch.ones(num_pos_edges,
```

Comment:
Mh, I am actually in favor of keeping it that way since `BCELoss` expects floating-point inputs. Does that work for you?

Reply:
I see, in this case do you think the assert below on line 36 should be removed?

Reply:
But yes generally fine by me, just wanted to make sure the behaviour was consistent

Reply:
I think the assert is a good check, since in case edge labels are given, we expect that this is a multi-class classification. If it is not, edge labels should not be passed to the loader in the first place.

Reply:
right, okay I'm sold on your idea - I'll revert this and update the docstring a bit